samcli/lib/bootstrap/bootstrap.py

```diff
@@ -83,6 +66,42 @@ def _create_or_get_stack(cloudformation_client):
         raise ManagedStackError(str(ex))
 
 
+def _check_sanity_of_stack(stack):
+    tags = stack.get("Tags", None)
+    outputs = stack.get("Outputs", None)
+
+    # For some edge cases, stack could be in invalid state
+    # Check if stack information contains the Tags and Outputs as we expected
+    if tags is None or outputs is None:
+        stack_state = stack.get("StackName", None)
+        msg = (
+            f"Stack {SAM_CLI_STACK_NAME} is missing Tags and/or Outputs information and therefore not in a "
+            f"healthy state (Current state:{stack_state}). Failing as the stack was likely not created "
+            f"by the AWS SAM CLI"
+        )
+        raise UserException(msg)
+
+    # Sanity check for non-none stack? Sanity check for tag?
+    try:
+        sam_cli_tag = next(t for t in tags if t["Key"] == "ManagedStackSource")
```

Comment:
I am guessing you are getting first value here. Probably pass a default parameter to `next` if `ManagedStackSource` is missing.

Reply:
Nevermind I noticed late that you are catching StopIteration.

Reply:
This was the previous code actually, I just moved it another method :)